paddle/fluid/framework/ir/mkldnn/mkldnn_conv_bn_fuse_pass_tester.cc

```diff
@@ -0,0 +1,255 @@
+    for (int i = 0; i < tnsr->numel(); ++i) {
+      ptr[i] = dist(engine);
+    }
+  }
+
+  void CompareTensors(Tensor* tensor1, Tensor* tensor2) {
+    // check dims
+    for (int i = 0; i < tensor1->numel(); ++i) {
+      EXPECT_NEAR(tensor1->data<float>()[i], tensor2->data<float>()[i], 1e-3);
+    }
+  }
+
+ public:
+  void MainTest(bool is_elementwise_add) {
+    auto base_prog = BuildProgramDesc(is_elementwise_add);
+
+    std::unique_ptr<ir::Graph> graph(new ir::Graph(base_prog));
+    Scope scope;
+    auto place = paddle::platform::CPUPlace();
+    NaiveExecutor exe{place};
+
+    auto pass = PassRegistry::Instance().Get(
+        is_elementwise_add ? "conv_transpose_eltwiseadd_bn_fuse_pass"
+                           : "conv_transpose_bn_fuse_pass");
+    graph->SetNotOwned(kParamScopeAttr, &scope);
+
+    auto& prog = graph->OriginProgram();
+
+    exe.CreateVariables(prog, 0, true, &scope);
+    exe.CreateVariables(prog, 0, false, &scope);
+
+    exe.Prepare(&scope, prog, 0, false);
+
+    std::cout << GenScopeTreeDebugInfo(&scope);
+
+    auto* a_tensor = exe.FindTensor("a");
+    auto* weights_tensor = exe.FindTensor("weights");
+    auto* g_tensor = exe.FindTensor("g");
+
+    // Batch Norm
+    auto* bias_bn_tensor = exe.FindTensor("bias_bn");  // shift
+    auto* scale_tensor = exe.FindTensor("scale");
+    auto* mean_tensor = exe.FindTensor("mean");
+    auto* variance_tensor = exe.FindTensor("variance");
+
+    // mb1_ic24oc24_ih8oh16kh2sh2dh0ph0_iw80ow160kw2sw2dw0pw0 deconv
+    a_tensor->Resize({1, 24, 160, 160});
+    weights_tensor->Resize({24, 24, 2, 2});
+    g_tensor->Resize({24});
+
+    bias_bn_tensor->Resize({24});
+    scale_tensor->Resize({24});
+    mean_tensor->Resize({24});
+    variance_tensor->Resize({24});
+
+    FillTensorWithRandomData(a_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(g_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(weights_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(bias_bn_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(scale_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(mean_tensor, 1.0f, 2.0f, place);
+    FillTensorWithRandomData(variance_tensor, 1.0f, 2.0f, place);
+
+    exe.Run();
+
+    // Get result without IR passes applied
+    // Need to copy result over as the same scope is used in both executors
+    // so first result will be overwritten by second
+    auto* j_tensor = exe.FindTensor("j");
+    Tensor no_ir_result;
+    TensorCopy(*j_tensor, place, &no_ir_result);
+
+    graph.reset(pass->Apply(graph.release()));
+
+    // Get Program from graph
+    ProgramDesc optimized_prog;
+    auto graph2program_pass =
+        paddle::framework::ir::PassRegistry::Instance().Get(
+            "graph_to_program_pass");
+    graph2program_pass->SetNotOwned<paddle::framework::ProgramDesc>(
+        "program", &optimized_prog);
+    graph2program_pass->Apply(graph.release());
+
+    exe.Prepare(&scope, optimized_prog, 0, false);
```

Comment:
I have one question. Why do you need to execute two graphs in one exe? Is it because the one QAT test which failed run like this? 
Does this PR also solve the customer model issue? Thanks

Reply:
I used one Executor as it was not needed to have two. Two executors can be used as well.

This PR is fixing some issues with cache clearing that were discovered when working on customer issue. UT introduced here is also used to reproduce an issue, but there is no fix to customer issue yet.

Reply:
I have updated the PR with candidate fix to issue #24585 . Currently testing